### PR TITLE
Add new calendar item struct

### LINKED ISSUE #2789: Create the CalendarItem struct
Ref #2754 for an overview of what this struct is.

It should implement `ITimeEntryPrototype` so that it can be easily used to start new time entries.

Please introduce the struct and, should it need any methods, please add tests ensuring the behaviour.

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?

Adds a new struct, `CalendarItem`, that represents the data we want to display for each item in the calendar view

### Relationships

Closes #2789

## Why do we want this?

Because we need it 😛 

## How is it done?

It's just a struct with some properties, a constructor that accepts a lot of parameters, and another one that accepts a IThreadSafeTimeEntry

### Why not another way?

This solution is simple

### Side effects

None

## Review hints

It's a straightforward review 🙂

## :squid: Permissions

Only my fellow 📆 friends

### CODE REVIEW

**Comment by daividssilverio** [comment]
🤔 The issue says it has to implement ITimeEntryPrototype. Why is this not happening here?

File: Toggl.Foundation/Calendar/CalendarItem.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            Source = source;
+            StartTime = startTime;
+            Duration = duration;
+            Description = description;
+            Color = color;
+            TimeEntryId = timeEntryId;
+        }
+
+        public CalendarItem(IThreadSafeTimeEntry timeEntry)
+            : this(CalendarItemSource.TimeEntry, timeEntry.Start, TimeSpan.FromSeconds(timeEntry.Duration.Value),
```
[the remark this answers is not in the document]
Comment (juanlaube) [reply Toggl.Foundation/Calendar/CalendarItem.cs]: 👀

File: Toggl.Foundation/Calendar/CalendarItem.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        public DateTimeOffset EndTime => StartTime + Duration;
+
+        public string Description { get; }
+
+        public string Color { get; }
+
+        public long? TimeEntryId { get; }
+
+        public CalendarItem(CalendarItemSource source, DateTimeOffset startTime, TimeSpan duration, string description,
+            string color = "", long? timeEntryId = null)
```
[the remark this answers is not in the document]
Comment (juanlaube) [reply Toggl.Foundation/Calendar/CalendarItem.cs]: 👀

**Comment by juanlaube** [comment]
@daividssilverio because, in principle, we don't need that interface, this is enough to represent a calendar item.

**Comment by amulware** [comment]
As already asked here: https://github.com/toggl/mobileapp/issues/2789#issuecomment-409826866, why a struct?

**Comment by juanlaube** [comment]
The struct is simple and it just holds the values we need, I don't see the point of using an interface. Maybe @willsb has something else to add in favor of this struct 😄

**Pull request merged**